jina/executors/indexers/keyvalue.py

```diff
@@ -159,23 +179,27 @@ def delete(self, keys: Iterable[str], *args, **kwargs) -> None:
             self._delete(keys)
 
     def _delete(self, keys: Iterable[str]) -> None:
-        for key in keys:
-            self.write_handler.header.write(
-                np.array(
-                    tuple(np.concatenate([[key], HEADER_NONE_ENTRY])),
-                    dtype=[
-                        ('', (np.str_, self.key_length)),
-                        ('', np.int64),
-                        ('', np.int64),
-                        ('', np.int64),
-                    ],
-                ).tobytes()
-            )
-            self._size -= 1
+        with self.write_handler as write_handler:
+            for key in keys:
+                write_handler.header.write(
+                    np.array(
+                        tuple(np.concatenate([[key], HEADER_NONE_ENTRY])),
+                        dtype=[
+                            ('', (np.str_, self.key_length)),
+                            ('', np.int64),
+                            ('', np.int64),
+                            ('', np.int64),
+                        ],
+                    ).tobytes()
+                )
+                self._size -= 1
 
     def _query(self, key):
         pos_info = self.query_handler.header.get(key, None)
-        if pos_info is not None:
+        if (
```

Comment:
```suggestion
        if (pos_info and getattr(self.query_handler, 'body', None))
```

no?

Reply:
Well, the real check is to check when we assign `pos_info` if `header` exists. But now I will raise an exception if the file in `_ReadHandler` does not exist, and then this will not be needed anymore